### ISSUE TITLE
False positive with `UP007` and `ForwardRef` on python 3.10

### ISSUE THREAD

**Issue by Liam-DeVoe** [astral-sh/ruff#20883]
### Summary

Ruff reports `UP007` when running `ruff check --target-version py310 '/Users/tybug/Desktop/sandbox4.py'` on the following:

```python
from typing import ForwardRef, Union

def f(x: Union[int, ForwardRef("A")]):
    pass
```

However, this is unsound on 3.10, because `int | ForwardRef("A")` errors. This is sound on 3.11+.

I know you're not really supposed to instantiate `ForwardRef` yourself, so you might consider this a wontfix. This arose for me when manually creating a signature which includes a forward ref in one of its types. See https://github.com/HypothesisWorks/hypothesis/pull/4558#discussion_r2402907294 for context.

### Version

ruff 0.14.0 (beea8cdfe 2025-10-07)

**Comment by ntBre** [comment]
Thanks for the report! This seems related to https://github.com/astral-sh/ruff/issues/19144. I also found https://github.com/python/cpython/issues/89652.
